src/errors.rs

```diff
@@ -26,15 +27,15 @@ pub enum DeployError {
     Link(#[from] LinkError),
 
     /// An error occured encoding deployment parameters with the contract ABI.
-    #[error("error ABI ecoding deployment parameters")]
+    #[error("error ABI ecoding deployment parameters: {0}")]
     Abi(#[from] AbiError),
```

Comment:
Where does the `{0}` fed from?

Reply:
This is special semantics for `#[derive(Error)]`. It comes from the first element in the enum variant.
Here is some more in-depth documentation: https://docs.rs/thiserror/1.0.4/thiserror/